[5.0] docs(treelist, gantt): Add Column Menu documentation (#1711)
telerik/blazor-docs: components/gantt/gantt-tree/templates/column-chooser.md

---
title: Column Chooser
page_title: Gantt - Column Chooser Template

> description: Customize the content of the Column Chooser.

Reviewer: The description must contain around 150 symbols.

> description: Use the Column Chooser Template of the Gantt to control the rendering and customize the content of the Column Chooser

Reviewer: ```suggestion
description: Use the Column Chooser Template of the Gantt to control the rendering and customize the content of the Column Chooser.
```

slug: gantt-templates-column-chooser
tags: telerik,blazor,gantt,templates,column,menu
published: True
position: 40
---

# Column Chooser Template

> The `Column Chooser Template` lets you control the rendering of the [ColumnChooser]({%slug gantt-column-menu%}#column-chooser). It exposes a `context` object that contains a List of all columns in the Gantt. 

Reviewer: "Column Chooser Template" doesn't appear to be a piece of code so use the default formatting instead of `inline code`.
```suggestion
The Column Chooser Template lets you control the rendering of the [ColumnChooser]({%slug gantt-column-menu%}#column-chooser). It exposes a `context` object that contains a list of all columns in the Gantt. 
```

> When the Template is setup the list of columns, rendered by default, would not be present.

Reviewer: ```suggestion
When you set up the Column Chooser Template, the list of columns that are rendered by default will not be present.
```

> The Column Chooser Template lets you control the rendering of the [ColumnChooser]({%slug gantt-column-menu%}#column-chooser). It exposes a [`context`](/blazor-ui/api/telerik.blazor.components.ganttcolumnchoosermenu) object that contains a list of all columns in the Gantt. 

Reviewer 2: Perhaps there was a misunderstanding in the MS Teams chat.

```suggestion
The Column Chooser Template lets you control the rendering of the [ColumnChooser]({%slug gantt-column-menu%}#column-chooser). It exposes a [`context`](/blazor-ui/api/Telerik.Blazor.Components.ColumnMenuChooserTemplateContext) object that contains a list of all columns in the Gantt. 
```

> >caption Use the Template to provide custom rendering

Reviewer: ```suggestion
>caption Using the Template to provide custom rendering
```

````CSHTML

> @* Use the Template to render the list of columns and add some custom styles *@ 

Reviewer: ```suggestion
@* Use the Template to render the list of columns and add some custom styles. *@ 
```

        var random = new Random();

        for (int i = 1; i < 6; i++)
        {
            var newItem = new FlatModel()
                {
                    Id = LastId,
                    Title = "Employee  " + i.ToString(),
                    Start = new DateTime(2020, 12, 10 + i),
                    End = new DateTime(2020, 12, 11 + i),
                    PercentComplete = Math.Round(random.NextDouble(), 2)
                };

            Data.Add(newItem);
            var parentId = LastId;
            LastId++;

            for (int j = 0; j < 5; j++)
            {
                Data.Add(new FlatModel()
                    {
                        Id = LastId,
                        ParentId = parentId,
                        Title = "    Employee " + i + " : " + j.ToString(),
                        Start = new DateTime(2020, 12, 20 + j),
                        End = new DateTime(2020, 12, 21 + i + j),
                        PercentComplete = Math.Round(random.NextDouble(), 2)
                    });

                LastId++;
            }
        }

        base.OnInitialized();
    }
}
````

>caption The result from the code snippet above

> ![templated column chooser example](images/templates-column-chooser-example.png)

Reviewer: Always include the component name and the framework in the alt text.
```suggestion
![Templated Column Chooser example with the Gantt component for Blazor](images/templates-column-chooser-example.png)
```
